modules/sfp_countryname.py

```diff
@@ -0,0 +1,323 @@
+            countryCode = region_code_for_country_code(phoneNumber.country_code)
+        except:
+            # Region code not present in source phone number
+            self.debug("Skipped invalid phone number: " + srcPhoneNumber)
+            return None
+        return abbvCountryCodes[countryCode.upper()]
+    
+    # Detect name of country from TLD of domain name
+    def detectCountryFromTLD(self, srcDomain):
+        
+        # Get dictionary of TLD country codes and country names
+        tldCountryCodes = self.getCountryCodeDict()
+        
+        # Get dictionary of non country TLD codes
+        tldNonCountryCodes = self.getNonCountryCodesDict()
+
+        # Split domain into parts by '.'
+        # We can conclude that country TLDs are reserved 
+        domainParts = srcDomain.split(".")
+
+        # Search for country TLD in the domain parts - reversed
+        for part in domainParts[::-1]:
+            if part.upper() in tldCountryCodes.keys():
+                return tldCountryCodes[part.upper()]
+
+        # Search for non country TLD in the domain parts
+        for part in domainParts[::-1]:
+            if part.upper() in tldNonCountryCodes.keys():
+                return tldNonCountryCodes[part.upper()]
+
+        # No associated country name is found
+        return None
+
+    # Detect name of country from IBAN
+    def detectCountryFromIBAN(self, srcIBAN):
+
+        # Get dictionary of country codes and country names
+        tldCountryCodes = self.getCountryCodeDict()
+        try:
+            # Get country code from IBAN 
+            countryCode = srcIBAN[0:2].upper()
+            return tldCountryCodes[countryCode]
+        except:
+            # No country name is found in the IBAN
+            return None
+    
+    # Detect name of country from Who Is lookup data
+    def detectCountryFromWhoIs(self, srcWhoIs):
+        
+        # Get dictionary of country codes and  country names
+        abbvCountryCodes = self.getCountryCodeDict()
+        countries = set()
+
+        # Look for countrycodes and country in whois data
+        for countryName in abbvCountryCodes.values(): 
+
+            # Look for country name in whois data
+            matchCountries = re.findall("[\s,'\"]" + countryName + "[\s,'\"]", srcWhoIs, re.IGNORECASE)
+
+            if len(matchCountries) > 0:
+                # Get country name from first index of list
+                # Extract only the text part of the country code
+                matchCountry = matchCountries[0].strip(",").strip("'").strip("\"").strip()
+                countries.add(matchCountry)
+        
+        # If any countries are found
+        if len(countries) > 0:
+            return countries
+
+        return None
+            
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["IBAN_NUMBER", "PHONE_NUMBER", "AFFILIATE_DOMAIN_NAME",
+                "CO_HOSTED_SITE_DOMAIN", "DOMAIN_NAME", "SIMILARDOMAIN",
```

Comment:
I think SIMILARDOMAIN is probably too far a reach from scope, since by definition such a domain will exist on a different TLD and thus probably result in a lot of false positives. At the very least this should be a configurable option, defaulted to False (you have an option for cohosts and affiliates but not similar domains).